perl.h

```diff
@@ -5427,6 +5427,25 @@ EXTCONST char *const PL_phase_names[] = {
     "END",
     "DESTRUCT"
 };
+
+/*
+=for apidoc_section $utility
+
+=for apidoc Amd|char *const|perl_phase_name|enum perl_phase
+
+Returns the given phase's name as a NUL-terminated string.
+
+For example, to print a stack trace that includes the current
+interpreter phase you might do:
+
+    const char* phase_name = perl_phase_name(PL_phase);
+    mess("This is weird. (Perl phase: %s)", phase_name);
+
+=cut
+*/
+
+#define perl_phase_name(phase) (PL_phase_names[phase])
+
 #else
 EXTCONST char *const PL_phase_names[];
```

Comment:
You'll need a no-op version of your macro here otherwise you'll break code.

Reply:
What would the macro do? Return NULL, empty-string, or … ?

It seems like in this case there’s no concept of a phase name, so anything that wants to access one should fail at compile time … but I’m probably misunderstanding something … ?

Reply:
I'm hoping you would have the answers to those questions.  Right now, this change would break things in a Perl where `DOINIT` is not defined.  There's probably some investigating to see where that is defined and what the differences are.

Reply:
Actually, the fix is to move the macro declaration out of that block.

That `DOINIT` check appears to be there so that perl.h works as well in Perl itself as in XS modules. In the former case we need to define the constant, whereas in the latter we just have to declare it (since Perl itself supplies the values).

Thank you!